src/playermat/Playmat.ttslua

```diff
@@ -620,12 +621,40 @@ function spawnTokensFor(object)
   tokenManager.spawnForCard(object, extraUses)
 end
 
-function onCollisionEnter(collisionInfo)
-  local object = collisionInfo.collision_object
+-- Modelled after onDeckResult from arkhamdb/ArkhamDb.ttslua, but with some necessary changes
+function spawnDeckFromToken(coin)
+  local deckData = JSON.decode(coin.getGMNotes())
+    destroyObject(coin)
+    deckData["playerColor"] = playerColor
+    -- Add Investigator Cards
+    deckData["slots"][deckData.investigatorId] = 1
+    deckData["slots"][string.sub(deckData.investigatorId, 1, 5) .. "-m"] = 1
+    -- Add Bonded Cards
+    for bondedId, bondedCount in pairs(deckData.bondedList) do
+      deckData.slots[bondedId] = bondedCount
+    end
+    -- Add Customizable Upgrade Sheets
+    
+    -- Add Summon Servitor minicard
+    if deckData.slots["09080"] ~= nil then
+      deckData.slots["09080-m"] = 1
+    end
+    -- Add On The Mend
+
+
+    deckImporterApi.loadCards(deckData)
+end
```

Comment:
This code should be part of the deckimporter instead of having it on each playmat